e2e/utils/waits-utils.ts

```diff
@@ -75,6 +75,23 @@ export async function waitForPropertyEquality(
   }
 }
 
+export async function waitForNumericalString(page: Page, xpath: string): Promise<string> {
+  await page.waitForXPath(xpath, {visible: true});
+
+  const numbers =  await page.waitForFunction( xpathSelector => {
+    const node = document.evaluate(xpathSelector, document.body, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null).singleNodeValue;
+    if (node !== null) {
+      const txt = node.textContent.trim();
+      const re = new RegExp(/^\d{1,3}(,\d{3})*(\.\d+)?$/);
```

Comment:
Cohort build page is not ready until Participants Count is calculated after adding new criterias. Participants Count numbers contain comma character(s). This function waits until Participants Count changes to a digits string with comma(s). It is used to get Counts and to indicate page is ready for next actions.

Reply:
This seems fragile. Participant counts may not always be over 1000. Could you instead check that the content changes?

Reply:
good catch. I updated regex to catch numbers less than one thousand. I need to check that calcualated count turns into a number.